### DISCUSSION TITLE
Client without `graphql` dependency

### DISCUSSION THREAD

**Discussion by AlecAivazis** [enisdenjo/graphql-ws#139] [Ideas]
Hey there!

I'm working on a client-side project that is trying to avoid any dependencies on the `graphql` package. I'm currently using `graphql-ws` but need to maintain a separate fork at the moment that removes the server implementation (and the dependency on `graphql`). 

Would you be interested in splitting `graphql-ws` into two separate packages? Something like `graphql-ws-client` and `graphql-ws-server`? They can both be exported from `graphql-ws` so that people can still depend on the combination but doing so would allow projects to avoid unnecessary dependencies. 

If this is something you are open to, I can submit a PR to get the process started :)

**Comment by enisdenjo** [comment]
Hey hey,

there is no need to remove the server part of the lib because it does not get bundled when targeting browsers. You'll only have what [`client.ts`](https://github.com/enisdenjo/graphql-ws/blob/master/src/client.ts) has (check the package.json). Do take a look [at the lib on Bundlephobia](https://bundlephobia.com/result?p=graphql-ws@4.2.1), since it bundles only client relevant code for browsers, it is only **2.2kB** in size minified + gzipped - which is pretty small IMHO. 😄


Having said this, splitting up the lib into 2 parts is completely unnecessary and is actually an overkill for a small lib like this one.

Furthermore, `graphql` is a peer-dependency, not a concrete one; it is indeed necessary because this is a GraphQL lib and we need it to be compliant and coherent with the spec. Another reason is that [both the Next and Error message depend on types from it](https://github.com/enisdenjo/graphql-ws/blob/d1421ad2ebe9ab81cbc7b3a1f83d0405eeb60a19/src/message.ts#L44-L54); those messages come from [the GraphQL over WebSocket Protocol](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md) and this lib is a reference implementation.

All use-cases that dont require GraphQL fall out of this lib's scope.

**Reply by AlecAivazis** [reply]
thanks for the reply! I didn't notice the `browser` target, that should address my needs perfectly
